presto-kafka07/src/main/java/com/facebook/presto/kafka/KafkaRecordSet.java

```diff
@@ -174,7 +178,10 @@ public boolean advanceNextPosition()
 
                 try {
                     // Create a fetch request
-                    openFetchRequest();
+                    KafkaFetchStatus kafkaFetchStatus = openFetchRequest();
+                    if (kafkaFetchStatus == KafkaFetchStatus.SAMPLE_ACHIEVED) {
```

Comment:
I think we can simply check if `messageAndOffsetIterator` is still `null` for this case so we can get rid of `KafkaFetchStatus` modification.